spring-integration-zeromq/src/main/java/org/springframework/integration/zeromq/outbound/ZeroMqMessageHandler.java

```diff
@@ -206,6 +246,47 @@ public void wrapTopic(boolean wrapTopic) {
 		this.wrapTopic = wrapTopic;
 	}
 
+	/**
+	 * Configure an URL for {@link org.zeromq.ZMQ.Socket#connect(String)}.
+	 * Mutually exclusive with the {@link #setBindPort(int)}.
+	 * @param connectUrl the URL to connect ZeroMq socket to.
+	 * @since 6.4
+	 */
+	public void setConnectUrl(@Nullable String connectUrl) {
+		this.setConnectUrl(() -> connectUrl);
+	}
+
+	/**
+	 * Configure an URL supplier for {@link org.zeromq.ZMQ.Socket#connect(String)}.
+	 * Mutually exclusive with the {@link #setBindPort(int)}.
+	 * @param connectUrl the supplier for URL to connect the socket to.
+	 * @since 6.4
+	 */
+	public void setConnectUrl(@Nullable Supplier<String> connectUrl) {
+		this.connectUrl = connectUrl;
+	}
+
+	/**
+	 * Configure a port for TCP protocol binding via {@link org.zeromq.ZMQ.Socket#bind(String)}.
+	 * Mutually exclusive with the {@link #setConnectUrl(String)}.
+	 * @param port the port to bind ZeroMq socket to over TCP.
+	 * @since 6.4
+	 */
+	public void setBindPort(int port) {
```

Comment:
We have these options covered by constructors.
I don't see a reason in extra setters on the matter: might cause some paradox of choice.